### PR TITLE
Add how-to on enabling LoadBalancer services on cloudscale.ch

### PR DESCRIPTION
**Pull request by glrf**

### CODE REVIEW

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
++
+[source,console]
+----
+whois -i mnt-by MNT-VSHN -T inetnum -r
+----
+Please coordinate with other VSHNeers before using an IP range.
+
+. Update RIPE DB to reflect the new usage of the IP range.
+
+. Create a ticket at cloudscale.ch to make the range available as a Floating Network in the project of the cluster.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:41]:
```suggestion
. Create a ticket with cloudscale.ch to make the range available as a Floating Network in the project of the cluster.
```

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+--
+
+[NOTE]
+--
+- In the future, this procedure will be mostly automated
+--
+
+== Starting situation
+
+* You already have a OpenShift 4 cluster on cloudscale.ch
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:15]:
```suggestion
* You already have an OpenShift 4 cluster on cloudscale.ch
```

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+        addresses:
+          - 203.0.113.128/25 <2>
+----
+<1> IP of the loadbalancers
+<2> Public IP range chosen
+
+. Update the Cilium configuration to enable BGP.
+Our Cilium setup is configured through the `CiliumConfig` CRD called `cilium` in namespace `cilium`.
+The CRD contains helm values.
+Update the CRD `spec` with
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:153]: Might be worth it to craft a `kubectl patch` which can be copy-pasted for this step

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+show ip route
+----
++
+This should include routes for the created LoadBalancer service.
+
+
+If these checks look correct, the BGP setup works as expected.
+If you still can't connect to the service, re-check the network policies and check if the Floating Network is assigned correctly.
+
+If the neighbors or routes don't show up correctly, follow the other troubleshooting steps.'
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:291]:
```suggestion
If the neighbors or routes don't show up correctly, follow the other troubleshooting steps.
``

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+In one of the Cilium pods run
++
+[source,console]
+----
+tcpdump -n -i any tcp port 179
+----
++
+If Cilium sends out announcements to the correct IPs, it's most likely setup correctly.
+If it doesn't, there is an issue with Cilium.
+One thing to consider it, that Cilium doesn't pick up updates of the `bgp-config` ConfigMap.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:307]:
```suggestion
One thing to consider is that Cilium doesn't automatically pick up updates of the `bgp-config` ConfigMap.
```

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+Make sure to restart the Cilium Pods it you change the configuration.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:308]: Which Cilium pods? Just the DaemonSet pods?

```suggestion
Make sure to restart the Cilium pods it you change the configuration.
```

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+[source,console]
+----
+tcpdump -n -i any tcp port 179
+----
++
+There should be packets coming in from the cluster nodes and they should be answered.
+
+.. If no packets come in, check the connection between the cluster nodes and the loadbalancer VM.
+
+.. If packets come in but aren't answered the issue might be the firewall setup.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:322]:
```suggestion
.. If packets come in but aren't answered, the issue might be the firewall setup.
```

**Review by simu — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+  neighbor <ip-infra-node-1> remote-as 64512
+  neighbor <ip-infra-node-2> remote-as 64512
+  end
+write <2>
+----
+<1> You can add any of the cluster node IPs.
+We propose to peer with each of the infrastructure nodes.
+<2> You need to write the configuration to persist 
+
+== Configure Cilium
```
[the remark this answers is not in the document]
Comment (glrf) [reply docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:118]: As the story for the Cilium component is in the next sprint, I don't think it's worth the effort.

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc (hunk 3 above)
Comment (glrf) [reply docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:154]: Again I don't know if it makes sense to improve this, as this should be managed by the Cilium component very soon.

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc [hunk 9]
```diff
@@ -0,0 +1,365 @@
+= Enable `LoadBalancer` serivces on cloudscale.ch
```
Comment (tobru) [inline docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:1]:
```suggestion
= Enable `LoadBalancer` services on cloudscale.ch
```

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc [hunk 10]
```diff
@@ -0,0 +1,365 @@
...
+
+[abstract]
+--
+Steps to enable `LoadBalancer` serivces on an OpenShift 4 cluster on https://cloudscale.ch[cloudscale.ch].
```
Comment (tobru) [inline docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:5]:
```suggestion
Steps to enable `LoadBalancer` services on an OpenShift 4 cluster on https://cloudscale.ch[cloudscale.ch].
```

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc [hunk 11]
```diff
@@ -0,0 +1,365 @@
...
+--
+
+[NOTE]
+--
+- In the future, this procedure will be mostly automated
```
Comment (tobru) [inline docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:10]: Remove that, doesn't add any value and when it gets automated, the documentation will get updated as well.

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc [hunk 12]
```diff
@@ -0,0 +1,365 @@
...
+== Prepare IP range
+
+. Get a suitable public IP range.
+You can find all IP ranges owned by VSHN
++
+[source,console]
+----
+whois -i mnt-by MNT-VSHN -T inetnum -r
+----
+Please coordinate with other VSHNeers before using an IP range.
```
Comment (tobru) [inline docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:37]: Who is "other VSHNeers"? We might need to be more specific here.

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc [hunk 13]
```diff
@@ -0,0 +1,365 @@
...
+
+. Update RIPE DB to reflect the new usage of the IP range.
```
Comment (tobru) [inline docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:39]: How to do that? We should say here what this actually means.

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc (hunk 8 above)
[the remark this answers is not in the document]
Comment (tobru) [reply docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:123]: I see that the same, no need to have an in-between solution right now.

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc [hunk 14]
```diff
@@ -0,0 +1,365 @@
...
+----
++
+This should include routes for the created LoadBalancer service.
+
+
+If these checks look correct, the BGP setup works as expected.
+If you still can't connect to the service, re-check the network policies and check if the Floating Network is assigned correctly.
+
+If the neighbors or routes don't show up correctly, follow the other troubleshooting steps.
+``
```
Comment (tobru) [inline docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:298]:
```suggestion

```

**Review by tobru — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc (hunk 12 above)
Comment (glrf) [reply docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:37]: The thing is: I'm not sure. It doesn't seem like there is really someone that feels responsible.

**Review by simu — APPROVED** [review]

File: docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc (hunk 12 above)
Comment (tobru) [reply docs/modules/ROOT/pages/how-tos/cloudscale/enable-loadbalancer-service.adoc:32]:
```suggestion
Please coordinate with VSHN Corporate IT before using an IP range.
```

**Review by tobru — APPROVED** [review]

**Pull request merged**